### PR TITLE
Client registry conformance improvements

### PR DESCRIPTION
**Pull request by FuriousLlama**
Improved logging and catching of errors if there is no PHN retrieved
If no PHN is found it wont attempt to call pharmanet services

### CODE REVIEW

**Comment by healthgatewaybc** [comment]
Medication Project [SonarQube Report](https://sonarqube-gateway.pathfinder.gov.bc.ca/dashboard?id=Medicationservice_pr_236)

File: Apps/Medication/src/Services/RestMedicationStatementService.cs [hunk 1]
```diff
@@ -94,6 +94,11 @@ private async Task<HNMessage<List<MedicationStatement>>> RetrieveMedicationState
                 string jwtString = this.httpContextAccessor.HttpContext.Request.Headers["Authorization"][0];
                 string phn = await this.patientDelegate.GetPatientPHNAsync(hdid, jwtString).ConfigureAwait(true);
 
+                if (string.IsNullOrEmpty(phn))
+                {
+                    return new HNMessage<List<MedicationStatement>>() { Result = Common.Constants.ResultType.Error, ResultMessage = "PHN could not be retrieved" };
```
Comment (sslaws) [inline Apps/Medication/src/Services/RestMedicationStatementService.cs]: I created an Error Messages constants class that should be used for error messages like this.

**Comment by healthgatewaybc** [comment]
WebClient Project [SonarQube Report](https://sonarqube-gateway.pathfinder.gov.bc.ca/dashboard?id=webclient_pr_236)

File: Apps/Medication/src/Services/RestMedicationStatementService.cs (hunk 1 above)
Comment (FuriousLlama) [reply Apps/Medication/src/Services/RestMedicationStatementService.cs]: good call. I've added that one to the constants and replaced it here.

**Comment by healthgatewaybc** [comment]
Medication Project [SonarQube Report](https://sonarqube-gateway.pathfinder.gov.bc.ca/dashboard?id=Medicationservice_pr_236)

**Pull request merged**